include/roaring/isadetection.h

```diff
@@ -207,14 +243,39 @@ static inline uint32_t croaring_detect_supported_architectures() {
 static inline bool croaring_avx2() {
   return false;
 }
+static inline bool croaring_avx512() {
+  return false;
+}
+#elif defined(__AVX512F__) && defined(__AVX512DQ__) && defined(__AVX512BW__) && defined(__AVX512VBMI2__) && defined(__AVX512BITALG__) && defined(__AVX512VPOPCNTDQ__)
+static inline bool croaring_avx2() {
+  return true;
+}
+static inline bool croaring_avx512() {
+  return true;
+}
 #elif defined(__AVX2__)
 static inline bool croaring_avx2() {
   return true;
 }
+static inline bool croaring_avx512() {
+  return false;
+}
 #else
 static inline bool croaring_avx2() {
   return  (croaring_detect_supported_architectures() & CROARING_AVX2) == CROARING_AVX2;
 }
+static inline bool croaring_avx512() {
+  static bool avx512_support = false;
```

Comment:
I am concerned that this code is setting avx512_support each and every time the function is called *if* avx512_support is false.

Why not have...

```C++
static bool avx512_support = (croaring_detect_supported_architectures() & (CROARING_AVX512F | CROARING_AVX512DQ | CROARING_AVX512BW | CROARING_AVX512VBMI2 | CROARING_AVX512BITALG | CROARING_AVX512VPOPCNTDQ)) 
 	  == (CROARING_AVX512F | CROARING_AVX512DQ | CROARING_AVX512BW | CROARING_AVX512VBMI2 | CROARING_AVX512BITALG | CROARING_AVX512VPOPCNTDQ));
return avx512_support;
```

In fact, the same strategy could be used in `croaring_avx2()`.

Reply:
yes, it is a bug for the first time. I am conernd that croaring_detect_supported_architectures() is time consuming each time this function is called.
i will change this code for your review.